src/Orchard.Web/Modules/Orchard.Layouts/Services/ElementManager.cs

```diff
@@ -79,7 +79,7 @@ orderby elementDescriptor.DisplayText.Text
 
         public ElementDescriptor GetElementDescriptorByTypeName(DescribeElementsContext context, string typeName) {
             var elements = DescribeElements(context);
-            var element = elements.SingleOrDefault(x => x.TypeName == typeName);
+            var element = elements.SingleOrDefault(x => String.Equals(x.TypeName, typeName, StringComparison.OrdinalIgnoreCase));
```

Comment:
Well, in this particular case, imagine a layout containing an element provided by Feature A. Now, Feature A is disabled, so when the layout is now loaded, the element pointing to the element descriptor provided by Layout A can no longer be found. I think this is a valid case for returning `null` rather than throwing an exception. hence, `SingleOrDefault` instead of `Single`.

But perhaps you meant something else.